homework-assignments-1/task-4.ts

```diff
@@ -0,0 +1,154 @@
+const ShippingCost: Record<string, number> = {
+    Tashkent: 10.8,
+    London: 15.7,
+    Toronto: 11.3,
+    Sydney: 14.7
+} as const
+
+const userLoginInfo: Record<string, string> = {
+    "user@gmail.com": "user123"
+} as const
+
+class Product {
+    constructor(public name: string, public price: number, public stock: number) { }
+
+    public addStock(quantity: number): void {
+        if (quantity > 0) this.stock += quantity;
+    }
+
+    public reduceStock(quantity: number): void {
+        if (this.stock >= quantity && quantity > 0) this.stock -= quantity;
+    }
+}
+
+class Customer {
+    constructor(public name: string, public email: string, public address: string) { }
+    public cart: Product[] = [];
+
+    public addToCart(product: Product): void {
+        if (product) this.cart.push(product);
```

Comment:
Instead of checking whether product exists, check is there any stock available if (product.getStock() > 0)